src/common/transformations/src/transformations/symbolic_transformations/dereshape_matmul.cpp

```diff
@@ -334,3 +335,68 @@ ov::pass::DeReshapeMatMul::DeReshapeMatMul() {
     auto m = std::make_shared<pattern::Matcher>(final_reshape, matcher_name);
     register_matcher(m, matcher_pass_callback);
 }
+
+ov::pass::DeReshapeFullyConnected::DeReshapeFullyConnected() {
+    MATCHER_SCOPE(DeReshapeFullyConnected);
+
+    auto reshaped_input = pattern::wrap_type<op::v1::Reshape>([](Output<Node> out) -> bool {
+        const auto& input_shape = out.get_node_shared_ptr()->get_input_partial_shape(0);
+        if (input_shape.rank().is_dynamic() || input_shape.size() < 3)
+            return false;
+        const auto& output_shape = out.get_partial_shape();
+        if (output_shape.rank().is_dynamic() || output_shape.size() < 2)
+            return false;
+        return dims_are_equal(input_shape[input_shape.size() - 1], output_shape[output_shape.size() - 1]);
+    });
+    auto converted =
+        pattern::wrap_type<op::v0::Convert>({reshaped_input}, pattern::consumers_count(1));  // optional convert
```

Comment:
just a note: it looks like we need to introduce pattern::optional<...> 
would it be useful?